config.hh.cmake

```diff
@@ -13,15 +13,13 @@
 # define @LIBRARY_NAME@_MINOR_VERSION @PROJECT_VERSION_MINOR_CONFIG@
 # define @LIBRARY_NAME@_PATCH_VERSION @PROJECT_VERSION_PATCH_CONFIG@
 
-// Since i <= 0 is always true, we'll get a compiler warning if major, minor or patch versions are equal to 0.
-// To avoid throwing an error on otherwise valid code, we explicitly perform a signed <= comparison
-#define @LIBRARY_NAME@_VERSION_AT_LEAST(major, minor, patch) (@LIBRARY_NAME@_MAJOR_VERSION>major || ((int)@LIBRARY_NAME@_MAJOR_VERSION>=(int)major && \
-                                                             (@LIBRARY_NAME@_MINOR_VERSION>minor || ((int)@LIBRARY_NAME@_MINOR_VERSION>=(int)minor && \
-                                                                                                     (int)@LIBRARY_NAME@_PATCH_VERSION>=(int)patch))))
+#define @LIBRARY_NAME@_VERSION_AT_LEAST(major, minor, patch) (@LIBRARY_NAME@_MAJOR_VERSION>major || (@LIBRARY_NAME@_MAJOR_VERSION>=major && \
+                                                             (@LIBRARY_NAME@_MINOR_VERSION>minor || (@LIBRARY_NAME@_MINOR_VERSION>=minor && \
+                                                                                                     @LIBRARY_NAME@_PATCH_VERSION>=patch))))
 
-#define @LIBRARY_NAME@_VERSION_AT_MOST(major, minor, patch) (@LIBRARY_NAME@_MAJOR_VERSION<major || ((int)@LIBRARY_NAME@_MAJOR_VERSION<=(int)major && \
-                                                            (@LIBRARY_NAME@_MINOR_VERSION<minor || ((int)@LIBRARY_NAME@_MINOR_VERSION<=(int)minor && \
-                                                                                                    (int)@LIBRARY_NAME@_PATCH_VERSION<=(int)patch))))
+#define @LIBRARY_NAME@_VERSION_AT_MOST(major, minor, patch) (@LIBRARY_NAME@_MAJOR_VERSION<major || (@LIBRARY_NAME@_MAJOR_VERSION<=major && \
+                                                            (@LIBRARY_NAME@_MINOR_VERSION<minor || (@LIBRARY_NAME@_MINOR_VERSION<=minor && \
+                                                                                                     @LIBRARY_NAME@_PATCH_VERSION<=patch))))
```

Comment:
`<=` should never warn, shouldn't it ?

Reply:
It warned with clang (I believe 12, maybe 10), I guess it optimised and flipped to >= during compilation (?)

Reply:
Warning happens at parsing stage. Optimization happens after so can't be flipped.

Reply:
Ok. It's obvious now. What warns is `<` and not `<=`.

Reply:
We can add a `@LIBRARY_NAME@_VERSION_EQUALS` macro and make `@LIBRARY_NAME@_VERSION_AT_MOST` rely on `@LIBRARY_NAME@_VERSION_AT_LEAST` and `@LIBRARY_NAME@_VERSION_EQUALS`

Reply:
That would make sense - thank you for the suggestions. For now, I've fixed it on the C++ side by converting the passed in values for major, minor, and patch to signed.